pkg/app/kafka.go

```diff
@@ -0,0 +1,444 @@
+	log.Info().Print("Waiting for application to be ready.", field.M{"app": kc.name})
+	ctx, cancel := context.WithTimeout(ctx, kafkaClusterWaitTimeout)
+	defer cancel()
+	err := kube.WaitOnDeploymentReady(ctx, kc.cli, kc.namespace, "strimzi-cluster-operator")
+	if err != nil {
+		return false, err
+	}
+	err = kube.WaitOnDeploymentReady(ctx, kc.cli, kc.namespace, "my-cluster-entity-operator")
+	if err != nil {
+		return false, err
+	}
+	err = kube.WaitOnStatefulSetReady(ctx, kc.cli, kc.namespace, "my-cluster-kafka")
+	if err != nil {
+		return false, err
+	}
+	err = kube.WaitOnStatefulSetReady(ctx, kc.cli, kc.namespace, "my-cluster-zookeeper")
+	if err != nil {
+		return false, err
+	}
+
+	log.Print("Application instance is ready.", field.M{"app": kc.name})
+	return true, nil
+}
+
+func (kc *KafkaCluster) Count(ctx context.Context) (int, error) {
+	log.Print("Counting records in kafka topic.", field.M{"app": kc.name})
+	count, err := consume(ctx, kc.topic, kc.namespace)
+	if err != nil {
+		return 0, errors.Wrapf(err, "Error counting the records for %s, %s.", kc.name, err)
+	}
+
+	log.Print("Count that we received from application is.", field.M{"app": kc.name, "count": count})
+	return count, nil
+}
+
+func (kc *KafkaCluster) Reset(ctx context.Context) error {
+	log.Print("Resetting the kafka topic is being handled in blueprint.", field.M{"app": kc.name})
+	return nil
+}
+
+// Initialize is used to initialize the database or create schema
+func (kc *KafkaCluster) Initialize(ctx context.Context) error {
+	return nil
+}
+
+// to produce messages to kafka
+func produce(ctx context.Context, topic string, namespace string) error {
+	partition := 0
+	forwarder, err := K8SServicePortForward(ctx, "my-cluster-kafka-external-bootstrap", namespace, "9094")
+	if err != nil {
+		return err
+	}
+
+	defer forwarder.Close()
+	ports, err := forwarder.GetPorts()
+	if err != nil {
+		return err
+	}
+	uri := "localhost:" + fmt.Sprint(ports[0].Local)
+
+	conn, err := kafka.DialLeader(ctx, "tcp", uri, topic, partition)
+	if err != nil {
+		return err
+	}
+	err = conn.SetWriteDeadline(time.Now().Add(10 * time.Second))
+	if err != nil {
+		return err
+	}
+	_, err = conn.WriteMessages(
+		kafka.Message{Value: []byte("{'title':'The Matrix','year':1999,'cast':['Keanu Reeves','Laurence Fishburne','Carrie-Anne Moss','Hugo Weaving','Joe Pantoliano'],'genres':['Science Fiction']}")},
+	)
+	if err != nil {
+		return err
+	}
+
+	if err := conn.Close(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// LoadConfig returns a kubernetes client config based on global settings.
+func LoadConfig() (*rest.Config, error) {
+	// log.Print("Attempting to use InCluster config")
+	config, err := rest.InClusterConfig()
+	if err == nil {
+		return config, nil
+	}
+
+	homeConfig := filepath.Join(os.Getenv("HOME"), ".kube/config")
+	config, err = clientcmd.BuildConfigFromFlags("", homeConfig)
+	if err == nil {
+		return config, nil
+	}
```

Comment:
Please use `kube.NewClient()` method in `pkg/kube` to init K8s client
https://github.com/kanisterio/kanister/blob/master/pkg/kube/client.go#L54